### DISCUSSION TITLE
Error Code when participant ID already exists

### DISCUSSION THREAD

**Discussion by SummerLi-MF** [phax/phoss-smp#288] [Q&A] [answered]
Hi SMP team, 

Our team is trying to provide better error code for requests when registering with participant ID already exists, we tried to curl PUT /{ServiceGroupId} with with existing participant ID to SMP client and currently the status code returned is 500 internal server error. We are wondering if there is a better way to identify the error caused by duplicate ids?

Here is the curl command we used to call SMP client:
```
$ curl -H "Content-Type: application/xml" -H "Authorization: Bearer test-token" -d '<smp:ServiceGroup xmlns:smp="http://busdox.org/serviceMetadata/publishing/1.0/" xmlns:id="http://busdox.org/transport/identifiers/1.0/"><id:ParticipantIdentifier scheme="iso6523-actorid-upis">0188:xxxx</id:ParticipantIdentifier><smp:ServiceMetadataReferenceCollection/></smp:ServiceGroup>' -X PUT http://localhost:8080/iso6523-actorid-upis::0188:xxxx -v
Error creating ServiceGroup 'iso6523-actorid-upis::0188:xxxx'

* Host localhost:8080 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
*   Trying [::1]:8080...
* Connected to localhost (::1) port 8080
> PUT /iso6523-actorid-upis::0188:xxxx HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/8.5.0
> Accept: */*
> Content-Type: application/xml
> Authorization: Bearer test-token
> Content-Length: 290
>
< HTTP/1.1 500
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, must-revalidate, proxy-revalidate
< Vary: Accept-Encoding
< Content-Type: text/plain;charset=UTF-8
< Content-Length: 63
< Connection: close
```
As always, thanks for maintaining all the smp code and been really helpful for responding to all the discussions!

Best regards,
Summer

**Comment by phax** [comment, marked as answer]
Well, the SML error handling is currently not good, because the SML is not responding standard compliant. See e.g. https://github.com/phax/peppol-commons/issues/48

**Reply by SummerLi-MF** [reply]
Thanks for the prompt reply and appreciate all the hard work maintaining 🚀
